test/connection.test.ts

```diff
@@ -522,41 +505,74 @@ describe('Connection', function () {
       }), 'Error connecting: Unable to determine path exchange rate')
     })
 
-    it('should determine the exchange rate even if it gets temporary errors', async function () {
```

Comment:
Ill bring this back with the F08 retry logic update, those are likely the only errors we should be retrying for.

Reply:
Why not T00 or T04?

Reply:
How likely is it that a connector recovers from a T04 or T00 immediately to allow the exchange rate calculation to go through if we retry? 

If we retry on a T04 or T00 its likely to just fail again. 

Do we retry all packet amounts that got a T04 if there are no F08 errors and no packets made it through?

My feeling is that an F08 is informative so we can use it to make another attempt. If we are hitting T04 and T00's then the connection will likely fail on another attempt.

Reply:
I'm split on this one. I think the correct response to a temporary error is to retry it, otherwise it shouldn't be a temporary error. However, you're probably right that on the network _right now_, retrying those errors is somewhat unlikely to help.

Reply:
Retrying T04s is important for a case we experienced with the eth plugin. When the eth client in moneyd connects to a server for the first time, moneyd will trigger settlement immediately to prefund it. But since creating a paychan takes a long time and the client initially has 0 balance with the server, the server will just return T04s until the paychan is created, so it's useful that Stream handles that case.

Reply:
@kincaidoneil for that case you outlined how long are you expecting retries for.

Say we send the volley of 1, 1000, 1000000, 10000000000 and get back T04's. 
Then we have a few options: 
1. Immediately retry
2. Wait and then retry
3. Bail and say the connection cannot be established 

If we do 3 and then the stream is tried again would that paychan be funded or are you depending on the stream retrying if it gets T04's for a period of time while that paychan is funded?

Reply:
For now I'm going with option 3 where we bail and say the connection cannot be established. I'm not sure if the use case you outlined needs stream to wait for the T04's to be cleared. Stream needs moneyd to already be running before being initialized locally so really the stream should not be setup until moneyd is already connected and the paychan has been made.